### PR TITLE
Don't select the row when <a> tags or <button> tags are clicked.

### LINKED ISSUE #3457: Clicking links with event handlers while using clickToSelect option
See #2154 for the initial discussion. Eventually, we experienced an issue with the workaround presented there. It works fine for a normal link, but turns out problematic when you want to add some event handlers to it. For example, if you'd like to show an alert once the link gets clicked, the workaround breaks it by stopping propagation of the event. A simple and reasonable solution would involve extending this condition https://github.com/wenzhixin/bootstrap-table/blob/21ae6acc49fe0229597fa33793cb09e4753fbd2a/src/bootstrap-table.js#L1869-L1875
to check if the event target is neither link nor button. This way a checkbox gets selected when the corresponding row is clicked, unless the clicked element is a link/button. Thoughts?

(I'm creating the issue as our comments on #2154 are left without any response.)

### PR DESCRIPTION
**Pull request by jfly**
This fixes #3457.

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1911,7 +1911,8 @@
             that.trigger(e.type === 'click' ? 'click-row' : 'dbl-click-row', item, $tr, field);
 
             // if click to select - then trigger the checkbox/radio click
-            if (e.type === 'click' && that.options.clickToSelect && column.clickToSelect) {
+            var ignoreTags = [ "A", "BUTTON" ];
+            if (e.type === 'click' && that.options.clickToSelect && column.clickToSelect && !ignoreTags.includes(e.target.tagName)) {
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: Lower version browser does not support `includes`, use `$.inArray` method instead.

File: src/bootstrap-table.js [hunk 2]
```diff
@@ -1911,7 +1911,8 @@
             that.trigger(e.type === 'click' ? 'click-row' : 'dbl-click-row', item, $tr, field);
 
             // if click to select - then trigger the checkbox/radio click
-            if (e.type === 'click' && that.options.clickToSelect && column.clickToSelect) {
+            var ignoreTags = [ "A", "BUTTON" ];
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: I think this can be consider as table option.

File: src/bootstrap-table.js (hunk 1 above)
Comment (jfly) [reply src/bootstrap-table.js]: Sure thing.

File: src/bootstrap-table.js (hunk 2 above)
Comment (jfly) [reply src/bootstrap-table.js]: Ok! I implemented this with an optional callback `ignoreClickToSelectOn` which takes an element and returns true or false. If you like this idea, how should I update the docs? I see `docs/_i18n/en/documentation/table-options.md`, but there are also Spanish and Chinese translations I don't feel comfortable trying to update =)

File: src/bootstrap-table.js [hunk 3]
```diff
@@ -394,6 +394,10 @@
 
         customSort: $.noop,
 
+        ignoreClickToSelectOn: function(element) {
+            return $.inArray(element.tagName, [ "A", "BUTTON" ];
+        },
```
Comment (wenzhixin) [inline src/bootstrap-table.js]:
```
ignoreClickToSelectOn: function (element) {
    return $.inArray(element.tagName, ['A', 'BUTTON']);
},
```

Comment (jfly) [reply src/bootstrap-table.js]: Whoops, sorry for the typo, I didn't actually test it out. I've updated the PR (along with the documentation), and actually tested it out this time. Things should be working now!

File: docs/_i18n/en/documentation/table-options.md [hunk 4]
```diff
@@ -524,6 +524,17 @@ The table options are defined in `jQuery.fn.bootstrapTable.defaults`.
         <td>false</td>
         <td>True to select checkbox or radiobox when clicking rows.</td>
     </tr>
+    <tr>
+        <td>ignoreClickToSelectOn</td>
+        <td>data-ignore-click-to-select-on</td>
+        <td>Function</td>
+        <td>{}</td>
```
Comment (wenzhixin) [inline docs/_i18n/en/documentation/table-options.md]: The default value is `<code>{ return $.inArray(element.tagName, ['A', 'BUTTON']); }</code>`

Comment (jfly) [reply docs/_i18n/en/documentation/table-options.md]: Ahh, that's what the `{}` is, got it! I've updated the docs accordingly.

**Comment by wenzhixin** [comment]
Thanks @jfly! @djhvscf can you help to update the es docs?

**Pull request merged**